modules/apps/layout/layout-page-template-test/src/testIntegration/java/com/liferay/layout/page/template/service/test/LayoutPageTemplateEntryServiceTest.java

```diff
@@ -209,6 +229,89 @@ public void testAddMultipleLayoutPageTemplateEntries()
 			actualLayoutPageTemplateEntries.size());
 	}
 
+	@Test
+	public void testAddWidgetLayoutPageTemplateEntry() throws PortalException {
+		/*
+		There are four types of page template entries. TYPE_BASIC,
+		TYPE_DISPLAY_PAGE, TYPE_MASTER_LAYOUT, TYPE_WIDGET_PAGE
+
+		What we want to test here is TYPE_WIDGET_PAGE. This is a special type
+		since it has another model associated with it. When creating a page
+		template of this type, another object of type LayoutPrototype is also
+		created.
+
+		So this test should check that. Suggested steps for the tests are:
+
+		1 - Create a new page template entry of type TYPE_WIDGET_PAGE
+		2 - Check that it was created
+		3 - Check that the LayoutPrototype for it was also created
+
+		Tips and tricks
+
+		1 - Theres a property in LayoutPageTemplateEntry that's called getLayoutPrototypeId()'
+		2 - Service for LayoutPrototype is in kernel
+
+		Bonus points
+
+		1 - Create other tests to see if the group was specified correctly
+		2 - Check that the name of the layout prototype is consistent
+		3 - Test that you cannot add duplicate names
+		4 - We also have a method called addGlobalLayoutPageTemplateEntry that we can test
+
+		Overall: HAVE FUN
+		 */
+		LayoutPageTemplateEntry layoutPageTemplateEntry =
+			LayoutPageTemplateTestUtil.addWidgetLayoutPageTemplateEntry(
+				_layoutPageTemplateCollection.
+					getLayoutPageTemplateCollectionId());
+
+		LayoutPageTemplateEntry persistedLayoutPageTemplateEntry =
+			_layoutPageTemplateEntryPersistence.fetchByPrimaryKey(
+				layoutPageTemplateEntry.getLayoutPageTemplateEntryId());
+
+		Assert.assertEquals(
+			layoutPageTemplateEntry.getName(),
+			persistedLayoutPageTemplateEntry.getName());
+
+		Assert.assertNotNull(layoutPageTemplateEntry.getLayoutPrototypeId());
+	}
+
+	@Test
+	public void testAssertGroupOfWidgetLayoutPageTemplateEntry()
```

Comment:
@Tim-Cao, I thing that for now we can leave this out, until https://issues.liferay.com/browse/LPS-115365 is fixed. Then we can add more tests for the group (even though I mentioned to create this test). What we can do for the moment is move the assert to the previous test. What do you think?